spot-client/src/spot-remote/ui/views/share/share.js

```diff
@@ -0,0 +1,218 @@
+    };
+
+    /**
+     * Initializes a new {@code Share} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            autoPromptScreenshare: true
+        };
+
+        this._onGoToSpotRemoveView = this._onGoToSpotRemoveView.bind(this);
+        this._onStartWirelessScreenshare
+            = this._onStartWirelessScreenshare.bind(this);
+        this._onStopWirelessScreenshare
+            = this._onStopWirelessScreenshare.bind(this);
+    }
+
+    /**
+     * Automatically start the screensharing flow. It is assumed if
+     * {@code Share} is displayed for the first time it is to immediately
+     * start screensharing.
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        this._onStartWirelessScreenshare();
+    }
+
+    /**
+     * Navigates away from {@code Share} if not connected to a Spot-TV.
+     *
+     * @inheritdoc
+     */
+    componentDidUpdate() {
+        if (!this.props.isConnectedToSpot) {
+            this.props.dispatch(addNotification('error', 'Disconnected'));
+            this.props.history.push('/');
+        }
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        return (
+            <NoSleep>
+                <View name = 'share-view'>
+                    { this._renderSubView() }
+                    <ElectronDesktopPickerModal />
+                </View>
+            </NoSleep>
+        );
+    }
+
+    /**
+     * Redirects to the full Spot-Remove view, exiting share mode.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onGoToSpotRemoveView() {
+        this.props.history.push('/remote-control');
+    }
+
+    /**
+     * Returns the content to display within {@code Share}.
+     *
+     * @private
+     * @returns {ReactComponent}
+     */
+    _renderSubView() {
+        if (this.state.autoPromptScreenshare) {
+            return <LoadingIcon color = 'white' />;
+        }
+
+        if (this.props.isWirelessScreensharing) {
+            return (
+                <StopShare
+                    onStopScreenshare = { this._onStopWirelessScreenshare } />
+            );
+        }
+
+        return (
+            <ModeSelect
+                isScreenshareActive = { this.props.isScreenshareActiveRemotely }
+                isWirelessScreenshareSupported = { isWirelessScreenshareSupported() }
+                onGoToSpotRemoveView = { this._onGoToSpotRemoveView }
+                onStartWirelessScreenshare = { this._onStartWirelessScreenshare } />
+        );
+    }
+
+    /**
+     * Starts the wireless screensharing flow.
+     *
+     * @returns {void}
+     */
+    _onStartWirelessScreenshare() {
+        if (this.props.isScreenshareActiveRemotely) {
+            this.setState({ autoPromptScreenshare: false });
```

Comment:
What happens when Spot Remote disconnects ? What's the flow in that situation and will the user get auto prompted again ?

Reply:
If spot remote disconnects, componentDidUpdate should kick in and take the remote back to join code entry. If the user enters a code again, they would get prompted again yes, because triggering of the desktop picker is controlled by state local to the share view.

Reply:
sounds good